nous-capybara/nous-capybara-34b-openai/model/model.py

```diff
@@ -0,0 +1,110 @@
+from threading import Thread
+
+import torch
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    GenerationConfig,
+    TextIteratorStreamer,
+)
+
+MODEL_NAME = "NousResearch/Nous-Capybara-34B"
+
+DEFAULT_STREAM = False
+
+
+def _format_prompt(messages: list[dict], add_generation_prompt: bool = True) -> str:
+    formatted_prompts = []
+    for message in messages:
+        if message["role"] == "user":
+            formatted_prompts.append(f"USER: {message['content']}")
+        elif message["role"] == "assistant":
+            formatted_prompts.append(f"ASSISTANT: {message['content']}")
+        # Note: Capybara doesn't support system messages. See https://huggingface.co/NousResearch/Nous-Capybara-34B/discussions/8
+    if add_generation_prompt:
+        formatted_prompts.append("ASSISTANT:")
+
+    return "\n".join(formatted_prompts)
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self.model = None
+        self.tokenizer = None
+
+    def load(self):
+        self.model = AutoModelForCausalLM.from_pretrained(
+            MODEL_NAME,
+            device_map="auto",
+            torch_dtype=torch.float16,
+            trust_remote_code=True,
+        )
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            MODEL_NAME, trust_remote_code=True
+        )
+
+    def preprocess(self, request: dict):
+        generate_args = {
+            "max_new_tokens": 256,
```

Comment:
I think this is handled here--let me know if I'm misunderstanding: https://github.com/basetenlabs/truss-examples/blob/a00a8685002d720d66ba0dc84b5c63a8211c4df0/nous-capybara/nous-capybara-34b-openai/model/model.py#L60-L65

Reply:
Might be easier to just iterate over the keys in the request instead of the comprehension. Up to you though.

```python
for k, v in request.keys():
    if k in default_generate_args:
        default_generate_args[k] = v
request["generate_args"] = default_generate_args
```

Reply:
Updated to not use the comprehension; I think this is more readable